From c877672ff6ec11efc8cd55d4c156211b16cf0975 Mon Sep 17 00:00:00 2001
From: tblume <Thomas.Blume@suse.com>
Date: Tue, 2 Aug 2016 14:55:25 +0200
Subject: [PATCH] systemd-ask-password: make sure directory watch is started
 before cryptsetup (#3850)

The password directory watch should get ordered before cryptsetup to make sure
that the password for unlocking the crypt device gets prompted.
(cherry picked from commit 7633f8ef371a0992374956437fb7bb8189156b10)
---
 units/systemd-ask-password-console.path | 2 +-
 units/systemd-ask-password-wall.path    | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/units/systemd-ask-password-console.path b/units/systemd-ask-password-console.path
index 2949635fea..7899ae788f 100644
--- a/units/systemd-ask-password-console.path
+++ b/units/systemd-ask-password-console.path
@@ -11,7 +11,7 @@ Documentation=man:systemd-ask-password-console.service(8)
 DefaultDependencies=no
 Conflicts=shutdown.target
 After=plymouth-start.service
-Before=paths.target shutdown.target
+Before=paths.target shutdown.target cryptsetup.target
 ConditionPathExists=!/run/plymouth/pid
 
 [Path]
diff --git a/units/systemd-ask-password-wall.path b/units/systemd-ask-password-wall.path
index 95ec9bc8a0..a3ca617256 100644
--- a/units/systemd-ask-password-wall.path
+++ b/units/systemd-ask-password-wall.path
@@ -10,7 +10,7 @@ Description=Forward Password Requests to Wall Directory Watch
 Documentation=man:systemd-ask-password-console.service(8)
 DefaultDependencies=no
 Conflicts=shutdown.target
-Before=paths.target shutdown.target
+Before=paths.target shutdown.target cryptsetup.target
 
 [Path]
 DirectoryNotEmpty=/run/systemd/ask-password
